kernel-impl/src/main/java/org/sakaiproject/user/impl/BaseUserDirectoryService.java

```diff
@@ -1973,6 +1971,39 @@ public UserEdit newUser(String eid)
 		return u;
 	}
 
+	public void updateUserId(String id,String newEmail)
+	{
+		try {
+			List<String> locksSucceeded = new ArrayList<String>();
+			// own or any
+			List<String> locks = new ArrayList<String>();
+			locks.add(SECURE_UPDATE_USER_ANY);
+			locks.add(SECURE_UPDATE_USER_OWN);
+			locks.add(SECURE_UPDATE_USER_OWN_EMAIL);
+			locksSucceeded = unlock(locks, userReference(id));
+
+			if(locksSucceeded.contains(SECURE_UPDATE_USER_ANY) || locksSucceeded.contains(SECURE_UPDATE_USER_OWN_EMAIL )|| locksSucceeded.contains(SECURE_UPDATE_USER_OWN)) {
+				UserEdit user = m_storage.edit(id);
+				if (user == null) {
+					M_log.error("Unexpected error when trying to validate the user");
```

Comment:
You should have a better error message, such as `Can't find user {id} when trying the update email address.` In fact it may well be much better to have some way to return errors from this method so the UI can do something sensible as otherwise it can't tell if it worked or not.